mssqlcli/packages/special/commands.py

```diff
@@ -93,3 +104,251 @@ def list_indexes(mssqlcliclient, pattern, verbose):
         base_query = base_query.format(pattern='')
 
     return mssqlcliclient.execute_multi_statement_single_batch(base_query)
+
+
+@special_command('\\df', '\\df[+] [pattern]', 'List functions.')
+def list_functions(mssqlcliclient, pattern, verbose):
+    pattern = pattern.replace('"', '').split('.')[-1]
+    base_query = '''
+SELECT {cols}
+  FROM sys.sql_modules m
+INNER JOIN sys.objects o
+        ON m.object_id=o.object_id
+WHERE type_desc like '%function%' {pattern}
+'''
+
+    if verbose:
+        base_query = base_query.format(cols='name, type_desc', pattern='{pattern}')
+    else:
+        base_query = base_query.format(cols='name', pattern='{pattern}')
+    if pattern:
+        base_query = base_query.format(pattern='and name like \'%{0}%\''.format(pattern))
+    else:
+        base_query = base_query.format(pattern='')
+
+    return mssqlcliclient.execute_multi_statement_single_batch(base_query)
+
+
+@special_command('\\sf', '\\sf FUNCNAME', 'Show a function\'s definition.')
+def show_function_definition(mssqlcliclient, pattern, verbose):
+    if not pattern:
+        click.secho('FUNCNAME is required. Usage \'\\sf FUNCNAME\'.', err=True, fg='red')
+        return []
+    pattern = pattern.replace('"', '').split('.')[-1]
+    base_query = '''
+SELECT definition
+  FROM sys.sql_modules m
```

Comment:
I thought sys.sql_modules was introduced in SQL 2008, so please verify against SQL 2005.

Reply:
Can run against Sql 2005, checked

---
In reply to: [166814651](https://github.com/dbcli/mssql-cli/pull/160#discussion_r166814651) [](ancestors = 166814651)